### ISSUE TITLE
False positive in <script> tag

### ISSUE THREAD

**Issue by MoogyG** [htmllint/htmllint#221]
I don't see anything about <script> tag handling. I met an issue when javascript code contains brackets used in a comparison.

In my case I have this error :

```html
[ Issue {
    line: 15,
    column: 12,
    code: 'E023',
    data: { chars: '>', part: 'text', desc: 'an unescaped angle bracket' },
    rule: 'spec-char-escape' } ]
```

when I lint this :

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <title>TEST</title>
</head>
<body>
    
<h1>test</h1>

<script>
    if (30 > 0) {
    //...
    }
</script>
</body>
</html>
```

**Comment by moraxy** [comment]
I ran into the same problem with `<script>` and `<code>` tags in my HTML and adding the deceptively simple `"raw-ignore-regex": /( [<>] )/` to my options solved it. Kinda depends on your codestyles, in that you need to have spaces around angled brackets in your javascript and none in your HTML - but so far it works for me.

**Comment by mlochbaum** [comment]
A more sophisticated, but still hacky, setting for `raw-ignore-regex` is `/\<script\>[^]*?\<\/script\>/`, which will avoid checking the contents inside `<script>...</script>` (exact spelling) at all. However, I do hope to find a sensible solution to make htmlling handle embedded non-html stuff like Javascript and svgs gracefully.

**Comment by s7726** [comment]
Alright some time over at https://regex101.com/ yielded the following.
Unfortunately it ignores the whole script tag part, not just the stuff in the middle.
```
"raw-ignore-regex": "/(\\<script.*?\\>\\<\/script\\>|\\<script.*?\\>[\\w\\W]+?\\<\/script\\>)/"
```
Lots of extra escaping
